crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -13,91 +13,107 @@
 // from outside) so it is only at that point that we need to check the hashes individually to the
 // find the one that got mined (if any).
 
-use super::{flashbots_api::FlashbotsApi, ESTIMATE_GAS_LIMIT_FACTOR};
+use super::ESTIMATE_GAS_LIMIT_FACTOR;
 use crate::settlement::Settlement;
 use anyhow::{anyhow, ensure, Context, Error, Result};
 use contracts::GPv2Settlement;
-use ethcontract::{transaction::Transaction, Account};
+use ethcontract::{contract::MethodBuilder, dyns::DynTransport, transaction::Transaction, Account};
 use futures::FutureExt;
 use gas_estimation::{EstimatedGasPrice, GasPriceEstimating};
 use primitive_types::{H256, U256};
 use shared::Web3;
-use std::time::{Duration, Instant, SystemTime};
+use std::time::{Duration, Instant};
 use web3::types::TransactionReceipt;
 
-pub struct FlashbotsSolutionSubmitter<'a> {
+/// Parameters for transaction submitting
+#[derive(Clone, Default)]
+pub struct SubmitterParams {
```

Comment:
I still think this is a bit of a magic toolbox with a bunch of unrelated things in it. `gas_price_cap`, `additional_miner_tip`, `retry_interval` and `target_confirm_time` should be passed into the submitter directly in my opinion.

The only thing that is dependent on the current submission is the gas_estimate and the deadline.

Reply:
I slightly agree, since we create a `Submitter::new`for every transaction, the division seems rather arbitrary to me and would be in favour of just shoving everything into one struct.

Of this seperation makes more sense for future refactor ideas, I'm also fine with it as it is.

Reply:
`SubmitterParams` is now reduced. However, I do have a feeling we will need params separated in the following refactor PRs, when we will define submitter but call it with different params for different strategies (maybe I'm wrong). Any way, I will remember this comment and revisit it later.